packages/flutter/test/gestures/tap_test.dart

```diff
@@ -398,10 +397,74 @@ void main() {
       'tapA onTapDown',
       'tapA onTapUp',
       'tapA onTap',
-      'tapB onTapCancel',
       'swept 2',
       'disposed A',
       'disposed B',
     ]);
   });
+
+  testGesture('PointerCancelEvent cancels tap', (GestureTester tester) {
+    const PointerDownEvent down = PointerDownEvent(
+        pointer: 5,
+        position: Offset(10.0, 10.0)
+    );
+    const PointerCancelEvent cancel = PointerCancelEvent(
+        pointer: 5,
+        position: Offset(10.0, 10.0)
+    );
+
+    final TapGestureRecognizer tap = TapGestureRecognizer();
+
+    final List<String> recognized = <String>[];
+    tap.onTapDown = (_) {
+      recognized.add('down');
+    };
+    tap.onTapUp = (_) {
+      recognized.add('up');
+    };
+    tap.onTap = () {
+      recognized.add('tap');
+    };
+    tap.onTapCancel = () {
+      recognized.add('cancel');
+    };
+
+    tap.addPointer(down);
+    tester.closeArena(5);
```

Comment:
I'm confused about exactly why we're closing the arena and he canceling the route 5 seconds later. It would help if you could explain this part of the test in terms of user gestures and widget tree changes.